docs/manager/graphql-reference/strawberry-schema.graphql

```diff
@@ -181,12 +543,161 @@ type PullArtifactPayload {
 }
 
 type Query {
+  _entities(representations: [_Any!]!): [_Entity]!
+  _service: _Service!
   artifacts(filter: ArtifactFilter = null, orderBy: [ArtifactOrderBy!] = null, first: Int = null, after: String = null): ArtifactConnection!
   artifactGroups(filter: ArtifactFilter = null, orderBy: [ArtifactOrderBy!] = null, before: String = null, after: String = null, first: Int = null, last: Int = null): ArtifactGroupConnection!
   artifact(id: ID!): Artifact
   artifactGroup(id: ID!): ArtifactGroup
+  deployments(
+    filter: DeploymentFilter = null
+    orderBy: DeploymentOrderBy = null
+
+    """Returns the first n items from the list."""
+    first: Int = null
+
+    """Returns the items in the list that come after the specified cursor."""
+    after: String = null
+
+    """Returns the items in the list that come before the specified cursor."""
+    before: String = null
+
+    """Returns the items in the list that come after the specified cursor."""
+    last: Int = null
+  ): ModelDeploymentConnection!
+  deployment(id: ID!): ModelDeployment
+  revisions(
+    filter: ModelRevisionFilter = null
+    order: ModelRevisionOrder = null
+
+    """Returns the first n items from the list."""
+    first: Int = null
+
+    """Returns the items in the list that come after the specified cursor."""
+    after: String = null
+
+    """Returns the items in the list that come before the specified cursor."""
+    before: String = null
+
+    """Returns the items in the list that come after the specified cursor."""
+    last: Int = null
+  ): ModelRevisionConnection!
+  revision(id: ID!): ModelRevision
+  deploymentMetrics(id: ID!, filter: DeploymentMetricsFilter = null): [DeploymentMetrics!]!
+  replica(id: ID!): ModelReplica
+}
+
+type ReplicaManagement {
+  desiredReplicaCount: Int!
+  replicas: [ModelReplica!]!
+  autoScalingRules: [EndpointAutoScalingRuleNode!]!
+  resolveDesiredReplicaCount: Int!
+  resolveReplicas: [ModelReplica!]!
+  resolveAutoScalingRules: [EndpointAutoScalingRuleNode!]!
+}
+
+type ReplicaMetric {
+  replicaId: ID!
+  cpuUsage: Float!
+  memoryUsage: Float!
+  requestCount: Int!
+}
+
+type ResourceConfig {
+  resourceGroup: ScalingGroupNode!
+  resourceSlots: ResourceSlots!
+  resourceOpts: ResourceOpts
+}
+
+input ResourceConfigInput {
+  resourceGroup: ScalingGroupNodeInput!
+  resourceSlots: ResourceSlotsInput!
+  resourceOpts: ResourceOptsInput = null
+}
+
+type ResourceOpts {
+  shmem: String
+  extra: JSONString
+}
+
+input ResourceOptsInput {
+  shmem: String = null
+  extra: JSONString = null
 }
 
+type ResourceSlots {
+  cpu: Int!
+  mem: String!
+  extra: JSONString
+}
```

Comment:
I think we said it was difficult to deliver this in this format last time...

Reply:
You'll probably have to write the ResourceSlots themselves as a JSONString and parse them internally before passing them.